lightning/src/ln/onchaintx.rs

```diff
@@ -0,0 +1,705 @@
+					if *is_htlc {
+						bumped_tx.input[i].witness.push(pubkey.unwrap().clone().serialize().to_vec());
+					} else {
+						bumped_tx.input[i].witness.push(vec!(1));
+					}
+					bumped_tx.input[i].witness.push(witness_script.clone().into_bytes());
+					log_trace!(self, "Going to broadcast Penalty Transaction {} claiming revoked {} output {} from {} with new feerate {}...", bumped_tx.txid(), if !is_htlc { "to_local" } else if HTLCType::scriptlen_to_htlctype(witness_script.len()) == Some(HTLCType::OfferedHTLC) { "offered" } else if HTLCType::scriptlen_to_htlctype(witness_script.len()) == Some(HTLCType::AcceptedHTLC) { "received" } else { "" }, outp.vout, outp.txid, new_feerate);
+				},
+				&InputMaterial::RemoteHTLC { ref witness_script, ref key, ref preimage, ref amount, ref locktime } => {
+					if !preimage.is_some() { bumped_tx.lock_time = *locktime }; // Right now we don't aggregate time-locked transaction, if we do we should set lock_time before to avoid breaking hash computation
+					let sighash_parts = bip143::SighashComponents::new(&bumped_tx);
+					let sighash = hash_to_message!(&sighash_parts.sighash_all(&bumped_tx.input[i], &witness_script, *amount)[..]);
+					let sig = self.secp_ctx.sign(&sighash, &key);
+					bumped_tx.input[i].witness.push(sig.serialize_der().to_vec());
+					bumped_tx.input[i].witness[0].push(SigHashType::All as u8);
+					if let &Some(preimage) = preimage {
+						bumped_tx.input[i].witness.push(preimage.clone().0.to_vec());
+					} else {
+						bumped_tx.input[i].witness.push(vec![0]);
+					}
+					bumped_tx.input[i].witness.push(witness_script.clone().into_bytes());
+					log_trace!(self, "Going to broadcast Claim Transaction {} claiming remote {} htlc output {} from {} with new feerate {}...", bumped_tx.txid(), if preimage.is_some() { "offered" } else { "received" }, outp.vout, outp.txid, new_feerate);
+				},
+				&InputMaterial::LocalHTLC { .. } => {
+					//TODO : Given that Local Commitment Transaction and HTLC-Timeout/HTLC-Success are counter-signed by peer, we can't
+					// RBF them. Need a Lightning specs change and package relay modification :
+					// https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2018-November/016518.html
+					return None;
+				}
+			}
+		}
+		log_trace!(self, "...with timer {}", new_timer);
+		assert!(predicted_weight >= bumped_tx.get_weight());
+		Some((new_timer, new_feerate, bumped_tx))
+	}
+
+	pub(super) fn block_connected<B: Deref, F: Deref>(&mut self, txn_matched: &[&Transaction], claimable_outpoints: Vec<Vec<ClaimRequest>>, height: u32, broadcaster: B, fee_estimator: F) -> Vec<SpendableOutputDescriptor>
+		where B::Target: BroadcasterInterface,
+		      F::Target: FeeEstimator
+	{
+		let mut new_claims = Vec::new();
+		let mut aggregated_claim = HashMap::new();
+		let mut aggregated_soonest = ::std::u32::MAX;
+		let mut spendable_outputs = Vec::new();
+
+		// Try to aggregate outputs if they're 1) belong to same parent tx, 2) their
```

Comment:
This comment is wrong now - we may aggregate an HTLC claim which spent a commitment tx with a spend of the commitment tx. This is, of course, fine, as any observer doesn't learn anything from that behavior, but the comment should be fixed.

Reply:
Hmmm why comment is wrong ?  Spending a HTLC output and to_local/to_remote output that still spending same commitment tx, sorry your comment is unclear to me.